### PR TITLE
Adds tab for realm role details + update/save functionality

### PR DESCRIPTION
**Pull request by jenny-s51**
## Motivation

https://issues.redhat.com/secure/RapidBoard.jspa?rapidView=4966&view=detail&selectedIssue=APPDUX-562

## Brief Description

Adds role details tab and functionality to edit name and description of role.

Extends #207 

## Checklist:

- [x] Code has been tested locally by PR requester
- [x] User-visible strings are using the react-i18next framework (useTranslation)
- [ ] Help has been implemented
- [ ] axe report has been run and resulting a11y issues have been resolved
- [ ] Unit tests have been created/updated
- [x] Formatting has been performed via prettier/eslint
- [ ] Type checking has been performed via 'yarn check-types'

### CODE REVIEW

**Review by edewit — CHANGES_REQUESTED** [review]
Is this Component supposed to be used for new roles and editing existing roles? If so I think these changes break the new role usecase right?

File: src/realm-roles/RealmRoleDetails.tsx [hunk 1]
```diff
@@ -10,24 +11,65 @@ import {
...
 export const RolesForm = () => {
   const { t } = useTranslation("client-scopes");
-  const { register, errors } = useForm<RoleRepresentation>();
+  const { register, handleSubmit, errors, control, setValue } = useForm<
+    RoleRepresentation
+  >();
   const history = useHistory();
-
+  const [name, setName] = useState("");
+  const [, setDescription] = useState("");
```
[the remark this answers is not in the document]
Comment (jenny-s51) [reply src/realm-roles/RealmRoleDetails.tsx]: nice catch, thank you. deleted

**Comment by jenny-s51** [comment]
@edewit great question, I just reached out to Haley on Zulip to confirm the expected behavior from the design

**Review by mfrances17 — CHANGES_REQUESTED** [review]
Changes look good, functionality is working as expected when tested. One reproducible issue I found, console is throwing an error on first load of the realm role details page, does not seem to appear on subsequent loads. Should be investigated either as part of this PR or a separate issue should be filed:

```
Warning: React does not recognize the `isDisabled` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `isdisabled` instead. If you accidentally passed it from a parent component, remove it from the DOM element.
    in textarea (created by TextArea)
    in TextArea (created by Controller)
    in Controller (created by RolesForm)
    in div (created by FormGroup)
    in div (created by FormGroup)
    in FormGroup (created by RolesForm)
    in form (created by Form)
    in Form (created by FormAccess)
    in FormAccess (created by RolesForm)
    in section (created by TabContentBase)
    in TabContentBase (created by ForwardRef)
    in ForwardRef (created by Tabs)
    in Tabs (created by RolesForm)
    in section (created by PageSection)
    in PageSection (created by RolesForm)
    in RolesForm (created by SecuredRoute)
    in SecuredRoute (created by component)
    in component (created by Context.Consumer)
    in Route (created by App)
    in Switch (created by App)
    in main (created by Page)
    in div (created by Page)
    in Page (created by App)
    in Router (created by BrowserRouter)
    in BrowserRouter (created by App)
    in DataLoader (created by ServerInfoProvider)
    in ServerInfoProvider (created by AppContexts)
    in AlertProvider (created by AppContexts)
    in Help (created by AppContexts)
    in AccessContextProvider (created by AppContexts)
    in AppContexts (created by App)
    in App
    in RealmContextProvider
    in DataLoader (created by WhoAmIContextProvider)
    in WhoAmIContextProvider

```

**Comment by edewit** [comment]
@mfrances17 I know what that is, it's not caused by this PR but by the
`<FormAccess` component, I'll create a Jira for it #230

**Comment by edewit** [comment]
@jenny-s51 what I mean is that the name field should be disabled: https://marvelapp.com/prototype/f39d464/screen/72683271

File: src/realm-roles/RealmRoleDetails.tsx [hunk 2]
```diff
@@ -10,53 +11,111 @@ import {
...
-                  id="kc-name"
-                  name="name"
-                />
+                {name ? (
+                  <TextInput
+                    ref={register({ required: true })}
+                    type="text"
+                    id="kc-name"
+                    name="name"
+                    // defaultValue={name}
```
Comment (dlabaj) [inline src/realm-roles/RealmRoleDetails.tsx:95]: Remove unneeded code.

**Review by dlabaj — APPROVED** [review]
LGTM

**Pull request merged**